test/util.go

```diff
@@ -4,20 +4,53 @@ import (
 	"context"
 	"fmt"
 	"strings"
+	"time"
 
 	testcluster "github.com/mesosphere/ksphere-testing-framework/pkg/cluster"
 	networkutils "github.com/mesosphere/ksphere-testing-framework/pkg/utils/networking"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// timeTimeout defines the maximum time to wait within portForwardPodWithPrefix for a pod and its containers
+	// to be Running and Ready. If we hit the timeout, there are most likely issues with the addon itself.
+	timeTimeout = 300 * time.Second
+)
+
 func portForwardPodWithPrefix(cluster testcluster.Cluster, ns, prefix, port string) (int, chan struct{}, error) {
 	pod, err := findPodWithPrefix(cluster, ns, prefix)
 	if err != nil {
 		return 0, nil, fmt.Errorf("could not find pod with prefix %s: %s", prefix, err)
 	}
-	if pod.Status.Phase != corev1.PodRunning {
-		return 0, nil, fmt.Errorf("pod %s is not running, it's in phase %s", pod.Name, pod.Status.Phase)
+
+	// Wait for Pod running and its containers to be ready for some time or timeout!
+	timeout := time.After(timeTimeout)
+	tick := time.Tick(500 * time.Millisecond)
+	var containersReady bool
+	for {
+		select {
+		case <-timeout:
+			return 0, nil, fmt.Errorf("Timeout after %.0f minutes - Pod %s in phase %s, containers of it are overall not ready", timeTimeout.Minutes(), pod.Name, pod.Status.Phase)
+		case <-tick:
+			// Check first if Pod is NOT in Running phase
```

Comment:
```suggestion
			// Check first that Pod is in Running phase
```
Negations are harder to comprehend so it's favourable to leave them out if possible.

Reply:
👍